content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,449 @@
+
+8. Click **Next: Review** to skip the **Add Tags** screen.
+
+9. Type the name `oidc-federation-test-role` for the IAM role and click **Create role**. 
+
+## Add the SPIFFE ID to the IAM Role
+
+To allow the workload from outside AWS to access AWS S3, add the workload's SPIFFE ID to the IAM role. This restricts access to the IAM role to JWT SVIDs with the specified SPIFFE ID.
+
+1. Navigate to the AWS [Identity and Access Management (IAM) page](https://console.aws.amazon.com/iam/home#/home), logging in if necessary.
+
+2. Click **Roles** on the left, use the search field to find the `oidc-federation-test-role` IAM role that you created in the last section, and click the role.
+
+3. At the top of the **Summary** page, next to **Role ARN**, copy the role ARN into the clipboard by clicking the small icon at the end. Save the ARN in a file such as `oidc-arn.txt` for use in the testing section ([Part 4](#part-4-test-access-to-aws-s3)).
+
+4. Click the **Trust relationships** tab near the middle of the page and then click **Edit trust relationship**.
+
+5. In the JSON access control policy, add a condition line at the end of the `StringEquals` attribute to restrict access to workloads matching the workload SPIFFE ID that was assigned in the [Kubernetes Quickstart](/spire/try/getting-started-k8s/). The new line to add is:
+
+   ```json
+   "MY_DISCOVERY_DOMAIN:sub": "spiffe://example.org/ns/default/sa/default"
+   ```
+   But substitute your Discovery document FQDN for MY\_DISCOVERY\_DOMAIN. Add a comma at the end of the previous line after `"mys3"`. When finished, the JSON should look similar to the following if you used oidc-discovery.example.org for your OIDC discovery domain:
+
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Principal": {
+           "Federated": "arn:aws:iam::012345678901:oidc-provider/oidc-discovery.example.org"
+         },
+         "Action": "sts:AssumeRoleWithWebIdentity",
+         "Condition": {
+           "StringEquals": {
+             "oidc-discovery.example.org:aud": "mys3",
+             "oidc-discovery.example.org:sub": "spiffe://example.org/ns/default/sa/default"
+           }
+         }
+       }
+     ]
+   }
+   ```
+
+6. Click **Update Trust Policy**. This change to the IAM role takes a minute or two to propagate.
+
+## Create an AWS S3 Bucket and Test File
+
+Create a simple test file in an AWS S3 bucket to use for testing.
+
+1. Create a text file on your local computer called `test.txt` with the following single line:
+
+   ```console
+   oidc-tutorial file
+   ```
+
+   You'll upload this file to the S3 bucket momentarily.
+
+2. Navigate to the AWS [Amazon S3 page](https://s3.console.aws.amazon.com/s3/home), logging in if necessary.
+
+3. Click **Create bucket**.
+
+4. Under **Bucket name** type `oidc-federation-test-bucket`.
```

Comment:
This won't work, because S3 buckets must have globaly unqiue names (that is, no two users anywhere can have the same AWS S3 bucket name). You'll have to ask them to make up a name and use that throughout the tutorial.

Reply:
Oh wow, good catch. I am moving this S3 bucket creation section to the first section under Part 3 so it's more natural to have users substitute their actual S3 bucket name in the JSON blob later on. I'm also explaining this requirement here during bucket creation.